src/server/middleware/security/security.go

```diff
@@ -0,0 +1,64 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package security
+
+import (
+	"net/http"
+
+	"github.com/goharbor/harbor/src/common/security"
+	"github.com/goharbor/harbor/src/common/utils/log"
+	"github.com/goharbor/harbor/src/core/config"
+	"github.com/goharbor/harbor/src/internal"
+	"github.com/goharbor/harbor/src/server/middleware"
+)
+
+var (
+	generators = []generator{
+		&secret{},
+		&oidcCli{},
+		&idToken{},
+		&authProxy{},
+		&robot{},
+		&basicAuth{},
+		&session{},
+		&unauthorized{},
+	}
+)
+
+// security context generator
+type generator interface {
+	Generate(req *http.Request) security.Context
+}
+
+// Middleware returns a security context middleware that populates the security context into the request context
+func Middleware() middleware.Middleware {
```

Comment:
Suggest to change the return type of `Middleware` func from `middleware.Middleware` to `func(http.Handler) http.Handler` so that we don't need do type conversion when run this middleware in beego.